Tasks/GooglePlayReleaseV4/GooglePlay.ts

```diff
@@ -0,0 +1,489 @@
+
+        const bundleFileList: string[] = getBundles(action);
+        tl.debug(`Bundles: ${bundleFileList}`);
+        const apkFileList: string[] = getApks(action);
+        tl.debug(`APKs: ${apkFileList}`);
+
+        const shouldPickObb: boolean = tl.getBoolInput('shouldPickObbFile', false);
+
+        if (shouldPickObb && apkFileList.length === 0) {
+            throw new Error(tl.loc('MustProvideApkIfObb'));
+        }
+
+        if (action !== 'OnlyStoreListing' && bundleFileList.length === 0 && apkFileList.length === 0) {
+            throw new Error(tl.loc('MustProvideApkOrAab'));
+        }
+
+        const track: string = tl.getInput('track', true);
+
+        const shouldAttachMetadata: boolean = tl.getBoolInput('shouldAttachMetadata', false);
+
+        let changelogFile: string = null;
+        let languageCode: string = null;
+        let metadataRootPath: string = null;
+
+        if (shouldAttachMetadata) {
+            metadataRootPath = tl.getPathInput('metadataRootPath', true, true);
+        } else {
+            changelogFile = tl.getInput('changelogFile', false);
+            languageCode = tl.getInput('languageCode', false) || 'en-US';
+        }
+
+        // Advanced inputs
+
+        const updatePrioritySupplied: boolean = tl.getBoolInput('changeUpdatePriority');
+        const updatePriority: number = Number(updatePrioritySupplied ? tl.getInput('updatePriority', false) : 0);
+
+        const userFractionSupplied: boolean = tl.getBoolInput('rolloutToUserFraction');
+        const userFraction: number = Number(userFractionSupplied ? tl.getInput('userFraction', false) : 1.0);
+
+        const uploadMappingFile: boolean = tl.getBoolInput('shouldUploadMappingFile', false) && (action === 'SingleApk' || action === 'SingleBundle');
+        const mappingFilePattern: string = tl.getInput('mappingFilePath');
+
+        const changesNotSentForReview: boolean = tl.getBoolInput('changesNotSentForReview');
+
+        const releaseName: string = tl.getInput('releaseName', false);
+
+        const versionCodeFilterType: string = tl.getInput('versionCodeFilterType', false) || 'all';
+        let versionCodeFilter: string | number[] = null;
+        if (versionCodeFilterType === 'list') {
+            versionCodeFilter = getVersionCodeListInput();
+        } else if (versionCodeFilterType === 'expression') {
+            versionCodeFilter = tl.getInput('replaceExpression', true);
+        }
+
+        // Warn about unused inputs
+
+        switch (action) {
+            case 'MultiApkAab': warnIfUnusedInputsSet('bundleFile', 'apkFile', 'shouldUploadMappingFile', 'mappingFilePath'); break;
+            case 'SingleBundle': warnIfUnusedInputsSet('apkFile', 'bundleFiles', 'apkFiles'); break;
+            case 'SingleApk': warnIfUnusedInputsSet('bundleFile', 'bundleFiles', 'apkFiles'); break;
+            case 'OnlyStoreListing': warnIfUnusedInputsSet('bundleFile', 'apkFile', 'bundleFiles', 'apkFiles', 'track'); break;
```

Comment:
Could it make sense to convert actions into objects? It allows us to encapsulate all information about each action in one place and it will make code a little bit easier to read. Also, could it make sense to combine all input arguments into a single object?